### ISSUE TITLE
An error occurred while installing nokogiri (1.6.8), and Bundler cannot continue.

### ISSUE THREAD

**Issue by TimB0** [appium/ruby_lib#373]
Any help would be greatly appreciated.  I would blame this on nokogiri, but I have other Rails gems / projects that are using nokogiri without any issue.  

**El-Capitan
Ruby Version = 2.2.1 (tried on 2.3.1 also with the same result**

**My Gemfile is:**
source 'https://rubygems.org'
gem 'appium_lib', '~> 8.0.2'

**bundle install**
Fetching gem metadata from https://rubygems.org/
Fetching version metadata from https://rubygems.org/
Resolving dependencies...
Using awesome_print 1.7.0
Using json 1.8.3
Using mini_portile2 2.1.0
Using pkg-config 1.1.7
Using ffi 1.9.10
Using rubyzip 1.2.0
Using websocket 1.2.3
Using tomlrb 1.2.2
Using bundler 1.12.5
Installing nokogiri 1.6.8 with native extensions

Gem::Ext::BuildError: ERROR: Failed to build gem native extension.

    current directory: /Users/tboland/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/nokogiri-1.6.8/ext/nokogiri
/Users/tboland/.rbenv/versions/2.2.1/bin/ruby -r ./siteconf20160702-91074-1jzc1bs.rb extconf.rb --use-system-libraries
Using pkg-config version 1.1.7
checking if the C compiler accepts ... yes
checking if the C compiler accepts -Wno-error=unused-command-line-argument-hard-error-in-future... no
Building nokogiri using system libraries.
ERROR: cannot discover where libxml2 is located on your system. please make sure `pkg-config` is installed.
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--without-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/Users/tboland/.rbenv/versions/2.2.1/bin/$(RUBY_BASE_NAME)
	--help
	--clean
	--use-system-libraries
	--with-zlib-dir
	--without-zlib-dir
	--with-zlib-include
	--without-zlib-include=${zlib-dir}/include
	--with-zlib-lib
	--without-zlib-lib=${zlib-dir}/lib
	--with-xml2-dir
	--without-xml2-dir
	--with-xml2-include
	--without-xml2-include=${xml2-dir}/include
	--with-xml2-lib
	--without-xml2-lib=${xml2-dir}/lib
	--with-libxml-2.0-config
	--without-libxml-2.0-config
	--with-pkg-config
	--without-pkg-config
	--with-xslt-dir
	--without-xslt-dir
	--with-xslt-include
	--without-xslt-include=${xslt-dir}/include
	--with-xslt-lib
	--without-xslt-lib=${xslt-dir}/lib
	--with-libxslt-config
	--without-libxslt-config
	--with-exslt-dir
	--without-exslt-dir
	--with-exslt-include
	--without-exslt-include=${exslt-dir}/include
	--with-exslt-lib
	--without-exslt-lib=${exslt-dir}/lib
	--with-libexslt-config
	--without-libexslt-config

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /Users/tboland/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/extensions/x86_64-darwin-15/2.2.0-static/nokogiri-1.6.8/mkmf.log

extconf failed, exit code 1

Gem files will remain installed in /Users/tboland/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/nokogiri-1.6.8 for inspection.
Results logged to /Users/tboland/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/extensions/x86_64-darwin-15/2.2.0-static/nokogiri-1.6.8/gem_make.out
Using childprocess 0.5.9
An error occurred while installing nokogiri (1.6.8), and Bundler cannot continue.
Make sure that `gem install nokogiri -v '1.6.8'` succeeds before bundling.

**mkmf.log**
"clang -o conftest -I/Users/tboland/.rbenv/versions/2.2.1/include/ruby-2.2.0/x86_64-darwin15 -I/Users/tboland/.rbenv/versions/2.2.1/include/ruby-2$
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: int main(int argc, char **argv)
4: {
5:   return 0;
6: }
/* end */

"clang -I/Users/tboland/.rbenv/versions/2.2.1/include/ruby-2.2.0/x86_64-darwin15 -I/Users/tboland/.rbenv/versions/2.2.1/include/ruby-2.2.0/ruby/ba$
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: int main() {return 0;}
/* end */

"clang -I/Users/tboland/.rbenv/versions/2.2.1/include/ruby-2.2.0/x86_64-darwin15 -I/Users/tboland/.rbenv/versions/2.2.1/include/ruby-2.2.0/ruby/ba$
error: unknown warning option '-Werror=unused-command-line-argument-hard-error-in-future'; did you mean '-Werror=unused-command-line-argument'? [-$
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: int main() {return 0;}
/* end */

"clang -o conftest -I/Users/tboland/.rbenv/versions/2.2.1/include/ruby-2.2.0/x86_64-darwin15 -I/Users/tboland/.rbenv/versions/2.2.1/include/ruby-2$
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: int main(int argc, char **argv)
4: {
5:   return 0;
6: }
/* end */

package configuration for libxml-2.0
cflags:
ldflags:
libs: -lxml2

"clang -o conftest -I/Users/tboland/.rbenv/versions/2.2.1/include/ruby-2.2.0/x86_64-darwin15 -I/Users/tboland/.rbenv/versions/2.2.1/include/ruby-2$
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: int main(int argc, char **argv)
4: {
5:   return 0;
6: }
/* end */

package configuration for libexslt
cflags:
ldflags:
libs: -lexslt -lxslt -lxml2 -lz -lpthread -licucore -lm -lxml2

"clang -E -I/Users/tboland/.rbenv/versions/2.2.1/include/ruby-2.2.0/x86_64-darwin15 -I/Users/tboland/.rbenv/versions/2.2.1/include/ruby-2.2.0/ruby$
conftest.c:3:10: fatal error: 'libxml/xmlversion.h' file not found
#include <libxml/xmlversion.h>
         ^
1 error generated.
checked program was:
/* begin */
1: #include "ruby.h"
2:
3: #include <libxml/xmlversion.h>
/* end */

**gem_make.out**
current directory: /Users/tboland/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/gems/nokogiri-1.6.8/ext/nokogiri
/Users/tboland/.rbenv/versions/2.2.1/bin/ruby -r ./siteconf20160702-91074-1jzc1bs.rb extconf.rb --use-system-libraries
Using pkg-config version 1.1.7
checking if the C compiler accepts ... yes
checking if the C compiler accepts -Wno-error=unused-command-line-argument-hard-error-in-future... no
Building nokogiri using system libraries.
ERROR: cannot discover where libxml2 is located on your system. please make sure `pkg-config` is installed.
*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
        --with-opt-dir
        --without-opt-dir
        --with-opt-include
        --without-opt-include=${opt-dir}/include
        --with-opt-lib
        --without-opt-lib=${opt-dir}/lib
        --with-make-prog
        --without-make-prog
        --srcdir=.
        --curdir
        --ruby=/Users/tboland/.rbenv/versions/2.2.1/bin/$(RUBY_BASE_NAME)
        --help
        --clean
        --use-system-libraries
        --with-zlib-dir
        --without-zlib-dir
        --with-zlib-include
        --without-zlib-include=${zlib-dir}/include
        --with-zlib-lib
        --without-zlib-lib=${zlib-dir}/lib
        --with-xml2-dir
        --without-xml2-dir
        --with-xml2-include
        --without-xml2-include=${xml2-dir}/include
        --with-xml2-lib
        --without-xml2-lib=${xml2-dir}/lib
        --with-libxml-2.0-config
        --without-libxml-2.0-config
        --with-pkg-config
        --without-pkg-config
        --with-xslt-dir
        --without-xslt-dir
        --with-xslt-include
        --without-xslt-include=${xslt-dir}/include
        --with-xslt-lib
        --without-xslt-lib=${xslt-dir}/lib
        --with-libxslt-config
        --without-libxslt-config
        --without-libxslt-config
        --with-exslt-dir
        --without-exslt-dir
        --with-exslt-include
        --without-exslt-include=${exslt-dir}/include
        --with-exslt-lib
        --without-exslt-lib=${exslt-dir}/lib
        --with-libexslt-config
        --without-libexslt-config

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /Users/tboland/.rbenv/versions/2.2.1/lib/ruby/gems/2.2.0/extensions/x86_64-darwin-15/2.2.0-static/nokogiri-1.6.8/mkmf.log

extconf failed, exit code 1

**Comment by bootstraponline** [comment]
It's nokogiri, nothing to do with appium.

Try `gem install nokogiri`

**Comment by JaniJegoroff** [comment]
I came across with nokogiri issue in the other project. I solved it by installing nokogiri with command:
```
gem install nokogiri -- --with-xml2-include=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include/libxml2 --use-system-libraries
```
or if you need a specific version:
```
gem install nokogiri -v '1.6.6.2' -- --with-xml2-include=/Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX10.11.sdk/usr/include/libxml2 --use-system-libraries
```

**Comment by JaniJegoroff** [comment]
Closing since nothing to do with Appium.

**Comment by crsantos** [comment]
Hey, here's the exact same error: https://github.com/wpscanteam/wpscan/issues/876#issuecomment-170403692

**Comment by Thialyson** [comment]
(for UBUNTU) First try to do this

`sudo apt-get install build-essential patch ruby-dev zlib1g-dev liblzmadev`

then try to install.

**Comment by yanlee26** [comment]
```
ogiri-1.6.8/gems/nokogiri-1.6.8
for inspection.
Results logged to
/var/folders/7v/5q6qv12s6s7fz1x3tm7lltc80000gn/T/bundler20180127-54584-6scnu3nokogiri-1.6.8/extensions/universal-darwin-17/2.3.0/nokogiri-1.6.8/gem_make.out

An error occurred while installing nokogiri (1.6.8), and Bundler cannot
continue.
Make sure that `gem install nokogiri -v '1.6.8'` succeeds before bundling.

In Gemfile:
  github-pages was resolved to 89, which depends on
    jekyll-mentions was resolved to 1.1.3, which depends on
      html-pipeline was resolved to 2.4.2, which depends on
        nokogiri
```

Same issue here...
